guide/using-python.md

```diff
@@ -92,3 +92,49 @@ Open the DOS console (CMD.EXE) and type:
 PYTHON_PATH\Scripts\pip.exe install opencv-python
 ```
 where `PYTHON_PATH` is the path to the Python installation directory, for example `C:\Python27`.
+
+### Use an alternative Python version
+
+As described above, the Python libraries for Webots are precompiled with Python 2.7.
+It is possible to use another Python version such as Python 3 or macOS brew Python,
+by recompiling these libraries.
+This documentation is generic and may not cover all the cases.
+This task should be reserved for users having strong knowledge in their OS, build systems
+and in terminal management.
+
+#### Python 3.6 on Windows
+
+1. Uninstall any Python installation.
+2. Download and install [Python 3.6 for x86_64](https://www.python.org/downloads/):
+    - Extend your system `PATH` environment variable to add the
+    `C:\Users\fabien\AppData\Local\Programs\Python\Python36` directory.
+    - Add `extern double hypot(double, double);` at the 3rd line of
+    `C:\Users\fabien\AppData\Local\Programs\Python\Python36\include\Python.h`
+3. Download, install and run [MSYS2 for x86_64](http://www.msys2.org/).
+4. If Webots is installed in `C:\\Program Files` then the write rights are certainly disabled.
+You should enable the user write rights at least for these paths:
+    - `/C/Program\ Files/Webots/resources/languages/python`
+    - `/C/Program\ Files/Webots/lib/python`
+5. Replace every occurrence of `PyString_FromStringAndSize` by `PyString_FromStringAndSize` in
+`/C/Program\ Files/Webots/resources/languages/python/controller.i`.
+6. Remove the line `init_controller` in `/C/Program\ Files/Webots/resources/languages/python/_controller.def`.
```

Comment:
Why? Don't we need it?

Reply:
Swig generates the following lines in:

```
#if PY_VERSION_HEX >= 0x03000000
#  define SWIG_init    PyInit__controller
#else
#  define SWIG_init    init_controller
#endif
```

The entree point differ from Python 2 to 3. I can fix this in the Makefile to simplify the instructions.